Source/core/Proxy.h

```diff
@@ -588,13 +588,16 @@ namespace Core {
         {
             ASSERT(a_Index < m_Current);
             ASSERT(m_List != nullptr);
+            ASSERT(a_Entry.IsValid() != false);
```

Comment:
Perhaps I'm missing something but I think this ASSERT is incorrect.
a_Entry is only assigned to below. E.g. I think here it would always be triggered now : https://github.com/rdkcentral/Thunder/blob/eadbd1e71ba415ed04247e9e3314b3ad2e49682b/Source/core/Proxy.h#L659